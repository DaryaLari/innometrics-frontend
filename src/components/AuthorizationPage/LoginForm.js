import React from 'react'
import { Link } from 'react-router-dom'
import { Field } from 'redux-form'
import Button from '../Button'
import Input from '../Input'
import { email, required } from '../../helpers/formValidators'
import styles from './style.css'

class LoginForm extends React.Component {
  onSubmit = () => {
    this.props.login()
  }

  componentDidMount(){
    console.log("here")
  }
  render () {
    return (
      <form className={styles.form}
            onSubmit={this.props.handleSubmit(this.onSubmit)}
      >
        <h1>Login</h1>
        <Field
          name="email"
          component={Input}
          label="Email"
          placeholder="Email"
          type="email"
          validate={[required, email]}
        />
        <Field
          name="password"
          component={Input}
          label="Password"
          placeholder="Password"
          type="password"
          validate={required}
        />
        <Button name="Login"
                type="submit"
                style={{width: '100%'}}
                disabled={this.props.submitDisabled}
        />
        <p className={styles.formError}>{this.props.formError}</p>
        <p>Don&#39;t have account yet?&nbsp;
          <Link to="/register">Register</Link>
        </p>
      </form>
    )
  }
}

export default LoginForm;