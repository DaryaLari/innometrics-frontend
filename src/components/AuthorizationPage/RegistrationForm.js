import React from 'react'
import { Link } from 'react-router-dom'
import { Field } from 'redux-form'
import Button from '../Button'
import Input from '../Input'
import { email, required } from '../../helpers/formValidators'
import styles from './style.css'

class RegistrationForm extends React.Component {
  onSubmit = () => {
    this.props.register()
  }

  render () {
    return (
      <form className={styles.form}
            onSubmit={this.props.handleSubmit(this.onSubmit)}
      >
        <h1>Register</h1>
        <Field
          name="email"
          component={Input}
          label="Email"
          placeholder="Email"
          type="email"
          validate={[required, email]}
        />
        <Field
          name="name"
          component={Input}
          label="Name"
          placeholder="Name"
          type="text"
          validate={required}
        />
        <Field
          name="surname"
          component={Input}
          label="Surname"
          placeholder="Surname"
          type="text"
          validate={required}
        />
        <Field
          name="password"
          component={Input}
          label="Password"
          placeholder="Password"
          type="password"
          validate={required}
        />
        <Button name="Register"
                type="submit"
                style={{width: '100%'}}
                disabled={this.props.submitDisabled}
        />
        <p className={styles.formError}>{this.props.formError}</p>
        <p>Have account already?
          <Link to="/login">Login</Link>
        </p>
      </form>
    )
  }
}

export default RegistrationForm;