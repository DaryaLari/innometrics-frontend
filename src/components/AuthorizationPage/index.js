import React from "react";
import styles from "./style.css";

class AuthorizationPage extends React.Component {
  render() {
    return (
        <div className={styles.content}>
          <LoginForm/>
        </div>

    )
  }
}

class LoginForm extends React.Component {
  render() {
    return (
        <div className={styles.form}>
          <h1>Login</h1>
          <div className={styles.inputRow}>
            <input placeholder="email"/>
          </div>
          <div className={styles.inputRow}>
            <input placeholder="password"/>
          </div>
            <p>Don't have account yet?
            <a href="/register">Register</a>
          </p>
        </div>
    )
  }
}

class RegisterForm extends React.Component {
  render() {
    return (
        <div className={styles.form}>
          <h1>Register</h1>
          <div className={styles.inputRow}>
            <input placeholder="email"/>
          </div>
          <div className={styles.inputRow}>
            <input placeholder="password"/>
          </div>
          <p>Have account already?
            <a href="/login">Login</a>
          </p>
        </div>
    )
  }
}

export default AuthorizationPage;