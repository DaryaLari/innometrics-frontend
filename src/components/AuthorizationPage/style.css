.content {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.form {
    background-color: rgba(255, 255, 255, 1);
    padding: 0 var(--cell-size);
}

.inputRow {
    height: calc(var(--cell-size) / 2);
}

.submitBtn {
    height: calc(var(--cell-size)/2);
    width: var(--cell-size);
}

.errorMsg {
    color: rgba(255, 0, 0, 1)
}

.formError {
    composes: errorMsg;
    height: calc(var(--cell-size)/4);
    width: 100%;
    text-align: center;
}
