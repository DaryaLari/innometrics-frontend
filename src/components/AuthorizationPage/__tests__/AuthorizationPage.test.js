import React from 'react'
import createStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import AuthorizationPage from "../index"
describe('Test AuthorizationPage component', () => {

  it('Test shallow render AuthorizationPage component', () => {

    const mockStore = createStore([]);
    const store = mockStore({})

    const wrapper = shallow(
      <Provider store={store}>
        <AuthorizationPage />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })

})