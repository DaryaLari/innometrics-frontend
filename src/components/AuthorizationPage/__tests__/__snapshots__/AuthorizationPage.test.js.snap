// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AuthorizationPage component Test renders 'Logout confirmation' when user authenticated 1`] = `
<div>
  <div>
    <h2>
      Logout confirmation
    </h2>
    <p>
      You are logged in already. Would you like to keep the same session or login to another account?
    </p>
    <div>
      <button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        Stay the same
      </button>
      <button
        disabled={false}
        onClick={[Function]}
        style={Object {}}
      >
        Logout anyway
      </button>
    </div>
  </div>
</div>
`;

exports[`Test AuthorizationPage component Test renders LoginForm when user is not authenticated 1`] = `
<div>
  <div>
    Login
  </div>
</div>
`;

exports[`Test AuthorizationPage component Test renders RegistrationForm when user is not authenticated 1`] = `
<div>
  <div>
    Register
  </div>
</div>
`;

exports[`Test AuthorizationPage component Test shallow render AuthorizationPage component 1`] = `<Connect(ReduxForm) />`;
